### ISSUE TITLE
Blob storage list - missing functionality vs API and PowerShell

### ISSUE THREAD

**Issue by hensilva** [Azure/azure-cli#5826]
Blob list operations using the CLI (az storage blob list) are limited to 5000 results [https://docs.microsoft.com/en-us/cli/azure/storage/blob?view=azure-cli-latest#az_storage_blob_list](url)

PowerShell implements the -ContinuationToken that allows paging and iteration for more control over the results and more than 5000 blobs can be listed. [https://docs.microsoft.com/en-us/powershell/module/azure.storage/get-azurestorageblob?view=azurermps-5.5.0](url)

It seems the -ContinuationToken is possible because the API offers the Marker parameter and NextMarker response: [https://docs.microsoft.com/en-us/rest/api/storageservices/list-blobs](url)

**Comment by williexu** [comment]
@hensilva a standard `storage blob list` will list ALL blobs.
We do however, currently have a limitation, where if the `--num-results` parameter is used, we only return up to 5,000 blobs.

**Comment by hensilva** [comment]
@williexu So it seems doc is not clear. According to my understanding 5000 is the limit regardless of --num-results being used or not. Doc states that if request does not specify --num-results or specifies a value greater than 5000, server will return 5000.

--num-results
Specifies the maximum number of blobs to return, including all BlobPrefix elements. _If the request does not specify num_results or specifies a value greater than 5,000, the server will return up to 5,000 items_. Setting num_results to a value less than or equal to zero results in error response code 400 (Bad Request).

That's why the paging capability that the API exposes is missing in the CLI. In such scenario, results would be delivered in batches of up to 5000 per iteration.

**Comment by williexu** [comment]
Yes, the documentation is unclear; in the CLI, we do the paging that will get the blobs past the first 5000.
